dl/src/main/scala/com/intel/analytics/bigdl/dataset/DataSet.scala

```diff
@@ -211,16 +216,18 @@ class CachedDistriDataSet[T: ClassTag] private[dataset] (buffer: RDD[Array[T]])
   }).reduce(_ + _)
 
   protected var indexes: RDD[Array[Int]] = buffer.mapPartitions(iter => {
-    Iterator.single(RandomGenerator.shuffle((0 until iter.next().length).toArray))
```

Comment:
I think the old behavior should not be changed.

if(isInOrder) {
   new code
} else {
   old code
}